drivers/mdc/mdc_npcx.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2021 Google LLC.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_mdc
+
+#include <drivers/mdc.h>
+#include <string.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(mdc, CONFIG_MDC_LOG_LEVEL);
+
+/** Device config */
+struct mdc_npcx_config {
+	/** The address of the MDC register */
+	uintptr_t addr;
+	/** The size of the MDC register (Unit: bytes) */
+	size_t size;
+};
+
+#define DRV_CONFIG(dev) ((const struct mdc_npcx_config *)(dev)->config)
+#define DRV_MDC_REG(dev, off) (*((volatile uint8_t *)DRV_CONFIG(dev)->addr) + off)
+
+static int mdc_npcx_read(const struct device *dev, size_t offset, size_t size, uint8_t *out)
+{
+	const struct mdc_npcx_config *config = (const struct mdc_npcx_config *)(dev->config);
+	volatile uint8_t *reg_addr = config->addr;
+
+	if (size < 1 || offset + size > config->size) {
+		return -EFAULT;
+	}
+
+	memcpy(out, reg_addr + offset, size);
```

Comment:
I don't recommend that read/write a register with different sizes access (e.g., read 32-bit data from an 8-bit register). Can we just add API for Bootloader scratch get & set. So we can access these registers with byte-length.

Reply:
Done